aws/resource_aws_globalaccelerator_listener.go

```diff
@@ -212,17 +212,9 @@ func resourceAwsGlobalAcceleratorListenerUpdate(d *schema.ResourceData, meta int
 	}
 
 	// Creating a listener triggers the accelerator to change status to InPending
-	stateConf := &resource.StateChangeConf{
-		Pending: []string{globalaccelerator.AcceleratorStatusInProgress},
-		Target:  []string{globalaccelerator.AcceleratorStatusDeployed},
-		Refresh: resourceAwsGlobalAcceleratorAcceleratorStateRefreshFunc(conn, d.Get("accelerator_arn").(string)),
-		Timeout: 5 * time.Minute,
-	}
-
-	log.Printf("[DEBUG] Waiting for Global Accelerator listener (%s) availability", d.Id())
-	_, err = stateConf.WaitForState()
+	err = resourceAwsGlobalAcceleratorAcceleratorWaitForDeployedState(conn, d.Get("accelerator_arn").(string))
 	if err != nil {
-		return fmt.Errorf("Error waiting for Global Accelerator listener (%s) availability: %s", d.Id(), err)
+		return nil
```

Comment:
Did you mean to swallow the error here? We should also continue to return the error context for operators and code maintainers:

```suggestion
		return fmt.Errorf("error waiting for Global Accelerator (%s) deployment: %s", d.Get("accelerator_arn").(string), err)
```

Otherwise folks will receive potentially confusing messages such as:

```
Error applying: unexpected state (FAILED), wanted (DEPLOYED), lastError: xxx
```

Reply:
No, that was an error. We can just pass the `err`, since the formatted error is now coming from `resourceAwsGlobalAcceleratorAcceleratorWaitForDeployedState`